app.js

```diff
@@ -0,0 +1,80 @@
+const express = require("express");
+require("express-async-errors");
+
+const app = express();
+
+app.set("view engine", "ejs");
+app.use(require("body-parser").urlencoded({ extended: true }));
+
+require("dotenv").config(); // to load the .env file into the process.env object
+const session = require("express-session");
+const MongoDBStore = require("connect-mongodb-session")(session);
+const url = process.env.MONGO_URI;
+
+const store = new MongoDBStore({
+  // may throw an error, which won't be caught
+  uri: url,
+  collection: "mySessions",
+});
+store.on("error", function (error) {
+  console.log(error);
+});
+
+const sessionParms = {
+  secret: process.env.SESSION_SECRET,
+  resave: true,
+  saveUninitialized: true,
+  store: store,
+  cookie: { secure: false, sameSite: "strict" },
+};
+
+if (app.get("env") === "production") {
+  app.set("trust proxy", 1); // trust first proxy
+  sessionParms.cookie.secure = true; // serve secure cookies
+}
+
+app.use(session(sessionParms));
+const passport = require("passport");
+const passportInit = require("./passport/passportInit");  //registers our local Passport strategy, and sets up the serializeUser and deserializeUser functions onto the passport object
+
+passportInit();
+app.use(passport.initialize()); //sets up Passport to work with Express and sessions
+app.use(passport.session()); //which sets up an Express middleware that runs on all requests, checks the session cookie for a user id, and if it finds one, deserializes and attaches it to the req.user property
+
+app.use(require("connect-flash")()); // handles flash messages to inform the user of the results
+app.use(require("./middleware/storeLocals"));
+app.get("/", (req, res) => {
+  res.render("index");
+});
+app.use("/sessions", require("./routes/sessionRoutes"));
+
+// secret word handling
+const secretWordRouter = require("./routes/secretWord");
+app.use("/secretWord", secretWordRouter);
```

Comment:
This is an error.  You have two app.use statements for the "/secretWord" route.  In this case, the first (line 53) will take effect, and the second (line 56) will do nothing.  As a result, the route is not protected by the auth middleware.  At least, that's what I think it will do.  It is definitely an error.